src/jvm/main/org/jetbrains/kotlinx/lincheck/IdeaPlugin.kt

```diff
@@ -128,6 +136,24 @@ fun onThreadSwitchesOrActorFinishes() {}
 
 // ======================================================================================================== //
 
+internal fun runPluginReplay(
+    testCfg: ModelCheckingCTestConfiguration,
+    testClass: Class<*>,
+    scenario: ExecutionScenario,
+    validationFunction: Actor?,
+    stateRepresentationMethod: Method?,
+    invocations: Int,
+    verifier: Verifier
+) {
+    testCfg.createStrategy(testClass, scenario, validationFunction, stateRepresentationMethod).use { replayStrategy ->
```

Comment:
Why do we need to re-create the strategy? I'm not sure this approach is good for Trace Debugger.

Reply:
Because `eventId`-s are only collected in the trace when `ideaPlugin` flag is enabled. 
I think we should just collect them always, it just an integer counter, there should be no overhead if we enable it always. However, `beforeEvent(eventId)` should only be called when the `ideaPlugin` flag is enabled (so no changes here required).

Reply:
I don't see why we cannot collect them when collecting the trace without re-creating the strategy.